nninit.lua

```diff
@@ -270,4 +270,95 @@ nninit.sparse = function(module, tensor, sparsity)
   return module
 end
 
+--[[
+--  Aghajanyan, A. (2017)
+--  Convolution Aware Initialization
+--  arXiv preprint arXiv:1702.06295
+--]]
+nninit.convaware = function(module, tensor, options)
+  -- The author of the paper provided a reference implementation for Keras: https://github.com/farizrahman4u/keras-contrib/pull/60
+
+  local sizes = tensor:size()
+  if #sizes ~= 4 then
+    error("nninit.convaware only supports 2d convolutions, feel free to issue a pull request to extend this implementation")
+  end
+
+  -- Make sure that the signal library is available, which provides the Fourier transform
+  local requireSignal = function()
+    require('signal')
+  end
+  if pcall(requireSignal) == false then
+    error("nninit.convaware requires the signal library, please make sure to install it: https://github.com/soumith/torch-signal")
+  end
+
+  -- Store the sizes of the convolution tensor to make the implementation easier to read
+  local filterCount = sizes[1]
```

Comment:
Use multiple assignment on 1 line for these to save some space whilst preserving readability.